coderd/workspacesearch.go

```diff
@@ -0,0 +1,80 @@
+package coderd
```

Comment:
This being in a separate file could lead Coder's (😉) to expect the workspace searching route exists in this file. It'd be my preference for this to go in `workspaces.go` instead.